### PR TITLE
Support v2 package repository

### PR DESCRIPTION
**Pull request by astj**
Ref: https://mackerel.io/blog/entry/announcement/20170616

### CODE REVIEW

File: spec/mackerel-agent_spec.rb [hunk 1]
```diff
@@ -5,7 +5,12 @@
 end
 
 describe service('mackerel-agent') do
-  it { should be_enabled   }
+  it {
+    if os[:family] == 'debian' && os[:release] >= '9.0'
+      pending('It seems serverspec cannot detect service status on docker debian 9...')
```
Comment (astj) [inline spec/mackerel-agent_spec.rb]: I confirmed mackerel-agent service is enabled properly in Debian9 on VirtualBox.
```
vagrant@gemmyD9ChefTestvagrant01:/vagrant$ sudo systemctl status mackerel-agent
● mackerel-agent.service - mackerel.io agent
   Loaded: loaded (/lib/systemd/system/mackerel-agent.service; enabled; vendor preset: enabled)
   Active: inactive (dead)
     Docs: https://mackerel.io/
```

File: recipes/default.rb [hunk 2]
```diff
@@ -21,31 +21,52 @@
...
     gpgkey gpgkey_url if yum_cookbook_ver >= Gem::Version.new('3.0.0')
     description "mackerel-agent monitoring"
     url repo_url
     action :add
   end
 elsif platform?('debian') or platform?('ubuntu')
-  package_options = '--force-yes -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
+  package_options = '--yes -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
+  repo_url = 'http://apt.mackerel.io/debian/'
+  if platform_supports_v2_repository and node[:kernel][:machine] === 'x86_64'
```
Comment (astj) [inline recipes/default.rb]: When this cookbook is applied to Debian 8 installed v1 repository (by hand or old cookbook), it's replaced to v2 repository.
I think it's acceptable behavior, but should I add some options like `prefer_v1`/`prefer_v2` ?

**Comment by itchyny** [comment]
LGTM

**Comment by astj** [comment]
Thanks!

**Pull request merged**